src/wr_logic_teleop_science/src/TeleopScienceLogic.cpp

```diff
@@ -32,14 +32,12 @@ auto main(int argc, char** argv) -> int {
     auto turnTableControlL = n.subscribe("/hci/science/gamepad/axis/shoulder_l", MESSAGE_CACHE_SIZE,
             static_cast<boost::function<void(const std_msgs::Bool::ConstPtr&)>>(
                 [&turnTableMsg, &turnTablePositions, &canListenL, &setpoint](const std_msgs::Bool::ConstPtr& msg) {
-                    if(static_cast<bool>(msg->data)) {
-                        if(canListenL) {
-                            setpoint = (setpoint + turnTablePositions.size() - 1) % turnTablePositions.size();
-                            std_msgs::Float64 x;
-                            x.data = turnTablePositions.at(setpoint);
-                            turnTableMsg.publish(x);
-                            canListenL = false;
-                        }
+                    if(msg->data && canListenL) {
```

Comment:
`msg->data` should still be `static_cast` to `bool`, since the `data` member isn't *really* a normal Boolean variable.